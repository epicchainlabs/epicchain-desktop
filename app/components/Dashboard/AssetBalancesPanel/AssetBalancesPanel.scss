@import '../../../styles/animations';

.assetBalancesPanel {
  .centeredContent {
    display: flex;
    justify-content: space-between;
    flex-direction: column;
    padding: 0;
  }

  .header {
    display: flex;
    align-items: center;
    justify-content: space-between;

    .refresh {
      cursor: pointer;

      &,
      & path {
        color: #9599a2;
        fill: currentColor;
      }

      &.loading {
        animation: spin 2s linear infinite;
      }
    }
  }

  .assets {
    margin: 0 24px 10px 24px;
    display: flex;
    align-items: flex-start;
    justify-content: space-around;

    .asset {
      text-align: center;

      .quantity {
        font-weight: 400;
        margin: -8px 0 16px 0;
        font-family: var(--font-gotham-light);
        font-size: 40px;
        color: var(--base-text);
      }

      .value {
        color: #969aa0;
        font-size: 14px;
      }

      .change {
        margin-left: 10px;
        color: white;
        font-size: 14px;
        font-weight: bold;
        border-radius: 3px;
        padding: 1px 4px;

        &.increase {
          background-color: #5abf6b;
        }

        &.decrease {
          background-color: #ee6d66;
        }
      }
    }
  }

  .walletTotal {
    font-family: var(--font-gotham-medium);
    font-size: 24px;
    color: var(--base-text);
    text-align: center;
    line-height: 32px;
  }

  .totalValue {
    padding: 24px;
    display: flex;
    background: var(--dashboard-asset-panel-total-background);
    justify-content: space-between;
    align-items: center;
    font-size: 24px;
    font-weight: 500;
    padding: 12px 24px;
  }

  .label {
    font-size: 24px;
    font-weight: 400;
    text-transform: uppercase;
    font-family: var(--font-gotham-light);
    font-size: 24px;
    color: var(--base-text);
  }

  .assetName {
    opacity: 0.5;
    font-family: var(--font-gotham-bold);
    font-size: 13px;
    color: var(--dashboard-asset-panel-asset-name);
    text-align: center;
  }

  .claim {
    margin: 0 24px 12px;
    width: 350px;
    align-self: center;

    .claimButton {
      width: 100%;
      margin-bottom: 12px;
    }
  }
}
