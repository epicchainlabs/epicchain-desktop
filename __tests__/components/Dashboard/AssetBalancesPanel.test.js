import React from 'react'
import nock from 'nock'
import { merge } from 'lodash-es'
import { mount, shallow } from 'enzyme'
import { progressValues } from 'spunky'

import { createStore, provideStore, provideState } from '../../testHelpers'
import AssetBalancesPanel from '../../../app/components/Dashboard/AssetBalancesPanel'
import {
  DEFAULT_CURRENCY_CODE,
  DEFAULT_LANGUAGE,
  MAIN_NETWORK_ID,
  ASSETS,
} from '../../../app/core/constants'
import IntlWrapper from '../../../app/components/Root/IntlWrapper'

const { LOADED } = progressValues

const initialState = {
  spunky: {
    network: {
      batch: false,
      progress: LOADED,
      data: MAIN_NETWORK_ID,
      loadedCount: 1,
    },
    auth: {
      batch: false,
      progress: LOADED,
      data: {
        address: 'ANqUrhv99rwCiFTL6N1An9NH5UVkPYxTuw',
      },
      loadedCount: 1,
    },
    settings: {
      batch: false,
      progress: LOADED,
      data: {
        currency: DEFAULT_CURRENCY_CODE,
        tokens: [],
        language: DEFAULT_LANGUAGE,
      },
      loadedCount: 1,
    },
    prices: {
      batch: false,
      progress: LOADED,
      data: {
        NEO: 25.48,
        GAS: 18.1,
      },
      loadedCount: 1,
    },
    priceHistory: {
      batch: false,
      progress: LOADED,
      data: {
        [ASSETS.NEO]: [
          {
            close: 5,
          },
          ...new Array(29).fill({}),
          {
            close: 10,
          },
        ],
        [ASSETS.GAS]: [
          {
            close: 10,
          },
          ...new Array(29).fill({}),
          {
            close: 5,
          },
        ],
      },
      loadedCount: 1,
    },
    balances: {
      batch: false,
      progress: LOADED,
      data: {
        NEO: '100001',
        GAS: '1000.0001601',
      },
      loadedCount: 1,
    },
    claims: {
      batch: false,
      progress: LOADED,
      data: {
        total: '0.5',
      },
      loadedCount: 1,
    },
  },
  claim: {
    claimRequest: false,
    disableClaimButton: false,
  },
}

describe('AssetBalancesPanel', () => {
  beforeEach(() => {
    const response = [
      { symbol: 'NEO', price_usd: 24.5 },
      { symbol: 'GAS', price_usd: 18.2 },
    ]
    nock('https://api.coinmarketcap.com')
      .get('/v1/ticker/')
      .query({ limit: 0, convert: 'USD' })
      .reply(200, response, { 'Access-Control-Allow-Origin': '*' })
  })

  test('renders without crashing', () => {
    const store = createStore(initialState)
    const wrapper = shallow(<AssetBalancesPanel store={store} />)
    expect(wrapper).toMatchSnapshot()
  })

  test.only('correctly renders data from state', () => {
    const wrapper = mount(
      provideState(
        <IntlWrapper>
          <AssetBalancesPanel />
        </IntlWrapper>,
        initialState,
      ),
    )

    expect(wrapper.find('#neoWalletValue').text()).toEqual('$2,548,025.48')
    expect(wrapper.find('#gasWalletValue').text()).toEqual('$18,100.00')
    expect(wrapper.find('#walletTotal').text()).toEqual('$2,566,125.48')
    expect(wrapper.find('#amountNeo').text()).toEqual('100,001')
    expect(wrapper.find('#amountGas').text()).toEqual('1,000.0002')
    expect(wrapper.find('#priceChangeNeo').text()).toEqual('+100.00%')
    expect(wrapper.find('#priceChangeGas').text()).toEqual('-50.00%')
  })

  test('account data refreshes when refresh button is clicked', () => {
    const store = createStore(initialState)
    const wrapper = mount(
      provideStore(
        <IntlWrapper>
          <AssetBalancesPanel />
        </IntlWrapper>,
        store,
      ),
    )

    wrapper
      .find('#refresh')
      .hostNodes()
      .simulate('click')

    expect(store.getActions()).toContainEqual(
      expect.objectContaining({
        type: 'balances/ACTION/CALL',
        meta: expect.objectContaining({ id: 'balances' }),
      }),
    )
  })

  test('correctly renders data from state with non-default currency', () => {
    const state = merge(initialState, {
      spunky: {
        settings: { data: { currency: 'eur' } },
        prices: { data: { NEO: 1.11, GAS: 0.55 } },
      },
    })
    const wrapper = mount(provideState(<AssetBalancesPanel />, state))

    expect(wrapper.find('#neoWalletValue').text()).toEqual('€111,001.11')
    expect(wrapper.find('#gasWalletValue').text()).toEqual('€550.00')
    expect(wrapper.find('#walletTotal').text()).toEqual('€111,551.11')
  })
})
