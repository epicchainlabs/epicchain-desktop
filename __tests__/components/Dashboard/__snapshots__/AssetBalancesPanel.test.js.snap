// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetBalancesPanel renders without crashing 1`] = `
<withData(Connect(withProgress(withProgressComponents(Connect(withData(Connect(withProgress(ConditionalCallComponent))))))))
  currencyCode="usd"
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    t {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
